### ISSUE TITLE
Elements of equal width in flexbox

### ISSUE THREAD

**Issue by 4ndr1u5** [Microsoft/reactxp#132]
When I use flexbox, items within it are not of equal width.
Basically, the problem (and the solution) is described in this stackoverflow thread (https://stackoverflow.com/questions/25066214/flexbox-not-giving-equal-width-to-elements)

Working solution is to use flexBasis property and set to 0.5 (in case of two elements in one row), however RX does not have flexBasis property in Types.d.ts. Is there any reason for that and if yes, what are other possible solutions for the problem?

**Comment by 3ygun** [comment]
@4ndr1u5 Hey haven't tried this on ReactXP but at least in browser of the `flex: 1 0 0` will make all flex items equal. This can be achieved by `flex: 1` in ReactXP following the `flex: n` given on the styles page ([https://microsoft.github.io/reactxp/docs/styles.html#flexbox-style-attributes]()).

On the JS Fiddle in your stackoverflow post if you add `flex-basis: 0;` (similar to your `flex-basis: 0.5` remark) it appears to make everything equal width. As such the `flex: 1` should give you the same set-up in ReactXP. Hope it works!

**Comment by erictraut** [comment]
@4ndr1u5, Can you explain your problem in more detail - perhaps include some code?

You should be able to generate equal-sized children with "flex: 1" in ReactXP. It supports the following:
flex: 0 // implies flexGrow: 0, flexShrink: 0, flexBasis: auto
flex: -1 // implies flexGrow: 0, flexShrink: 1, flexBasis: auto
flex: n // (where n is positive) implies flexGrow: n, flexShrink: 0, flexBasis: 0

So, if you specify "flex: 1" for each of the children, they each get a flexBasis of 0, and each child will  expand to take up 1/n of the space (assuming n children).

The stackoverflow thread you cited correctly points out that you may not get equal sizes if flexBasis is "auto", but that's not the case here.

**Comment by 4ndr1u5** [comment]
I tried using "flex:1" in RX and in web it generates "flex: 1 1 auto;".
in your post you say that "flex 1" should imply "flex: 1 0 0", it would work great for me if it was so..

On the other hand if i specify flexBasis property, i can see "flex: 1 1 0px" which also works fine.

row style:
```
row: RX.Styles.createViewStyle({
    flex: 1,
    flexDirection: 'row'
  })
```
item style:
```
image: RX.Styles.createImageStyle({
    flex: 1,
    height: 380,
    // flexBasis: 0
  })


```
to sum up, problem is that flex : 1 implies "flex: 1 0 0" but actual result for some reason is "flex: 1 1 auto". 
does it really works as it is supposed to?.. or am I missing something?

**Comment by erictraut** [comment]
@4ndr1u5, you are correct. The implementation was not consistent with the interface. Thanks for reporting this problem. 

I tried changing the implementation to match the documented behavior, but this (not surprisingly) broke a bunch of existing code. I decided instead to modify the documented interface to match the current implementation. Here's the updated documentation:

flex: 0 // implies "flex 0 0 auto"
flex: n // (where n is negative) implies "flex 0 n auto"
flex: p // (where p is positive) implies "flex p 1 auto"

Since the current implementation provides no way to specify a flexBasis of 0, as you pointed out, I've also added the ability to override flexBasis, flexGrow, and flexShrink individually. I'm in the process of testing the change. If you want to review [the PR](https://github.com/Microsoft/reactxp/pull/134), you're welcome to do so. I'll merge it and republish once I confirm that everything looks OK.
